### PR TITLE
Add app.drey.Warp

### PR DESCRIPTION
**Pull request by felinira**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by felinira** [comment]
bot, build app.drey.Warp

File: app.drey.Warp.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+runtime-version: 42
+sdk: org.gnome.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: warp
+
+finish-args:
+  # X11 and Wayland
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline app.drey.Warp.yaml:12]: `--share=ipc` is needed too

File: app.drey.Warp.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  - --filesystem=xdg-download
+
+build-options:
+  append-path: /usr/lib/sdk/rust-stable/bin
+  build-args:
+    - --share=network
+  test-args:
+    - --socket=x11
+    - --share=network
+  env: {}
```
Comment (hfiguiere) [inline app.drey.Warp.yaml:30]:
```suggestion
  build-args:
    - --share=network
  test-args:
    - --socket=x11
    - --share=network
  env: {}
```

Neither is allowed / possible.

`env` syntax invalid.

File: app.drey.Warp.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+modules:
+- name: warp
+  buildsystem: meson
+  config-opts:
+    - -Dprofile=default
+  run-tests: true
```
Comment (hfiguiere) [inline app.drey.Warp.yaml:37]:
```suggestion
```

given what you seem to want for the test (network, x11), they can't run

File: app.drey.Warp.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+app-id: app.drey.Warp
+runtime: org.gnome.Platform
+runtime-version: 42
```
Comment (hfiguiere) [inline app.drey.Warp.yaml:3]:
```suggestion
runtime-version: '42'
```

it's yaml, you have to tell it is a string.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: app.drey.Warp.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+app-id: app.drey.Warp
```
Comment (hfiguiere) [inline app.drey.Warp.yaml:1]: the domain `drey.app` doesn't seem to exist, so this make the app-id fail to pass requirements.

Comment (felinira) [reply app.drey.Warp.yaml:2]: It does exist. See [app.drey](https://gitlab.gnome.org/sophie-h/app.drey/)

**Comment by felinira** [comment]
bot, build app.drey.Warp

File: app.drey.Warp.yaml (hunk 5 above)
Comment (hfiguiere) [reply app.drey.Warp.yaml:2]: it surely doesn't resolve in the browser. Also you link to a gitlab repository, so no.

See
https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: app.drey.Warp.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  - --share=network
+  # Save files in the Downloads directory
+  - --filesystem=xdg-download
+build-options:
+  append-path: /usr/lib/sdk/rust-stable/bin
+  build-args:
+    - --share=network
+  test-args:
+    - --socket=x11
+    - --share=network
```
Comment (hfiguiere) [inline app.drey.Warp.yaml:26]:
```suggestion
```

I think my suggestion was unfinished. I really mean to remove the whole block.

Comment (felinira) [reply app.drey.Warp.yaml]: Ah sorry, I misunderstood that.

**Comment by felinira** [comment]
bot, build app.drey.Warp

**Pull request closed without merging**